lightningd/plugin.c

```diff
@@ -486,31 +502,51 @@ static bool plugin_opt_add(struct plugin *plugin, const char *buffer,
 		return false;
 	}
 
-	/* FIXME(cdecker) Support numeric and boolean options as well */
-	if (!json_tok_streq(buffer, typetok, "string")) {
-		plugin_kill(plugin,
-			    "Only \"string\" options currently supported");
-		return false;
-	}
-
 	popt = tal(plugin, struct plugin_opt);
+	popt->value = talz(popt, struct plugin_opt_value);
+	popt->value->as_str = talz(popt->value, char);
+	popt->value->as_int = talz(popt->value, int);
```

Comment:
Possibly better to leave `as_int` and `as_bool` as `NULL` if they are not of the appropriate type, and modify the earlier code that assigns them to only allocate at those points.